### PR TITLE
Update TC_OCC_2_1.py

### PR DESCRIPTION
**Pull request by jaehs6sam**
Updating TC 2.1 to address issue (https://github.com/project-chip/connectedhomeip/issues/35318)

### CODE REVIEW

File: src/python_testing/TC_OCC_2_1.py [hunk 1]
```diff
@@ -121,25 +121,31 @@ async def test_TC_OCC_2_1(self):
             hold_time_limits_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTimeLimits)
             asserts.assert_less_equal(hold_time_limits_dut.holdTimeMin, hold_time_limits_dut.holdTimeMax,
                                       "HoldTimeMin is not in valid range")
-            asserts.assert_greater_equal(hold_time_limits_dut.holdTimeMin, 0, "HoldTimeMin is not in valid range")
-            asserts.assert_less_equal(hold_time_limits_dut.holdTimeMax, 0xFFFE, "HoldTimeMin is not in valid range")
-            asserts.assert_greater_equal(hold_time_limits_dut.holdTimeMax,
-                                         hold_time_limits_dut.holdTimeMin, "HoldTimeMin is not in valid range")
+            asserts.assert_greater_equal(hold_time_limits_dut.holdTimeMin, 1, "HoldTimeMin minimum value is not 1.")
+            asserts.assert_greater_equal(hold_time_limits_dut.holdTimeMax, 10, "HoldTimeMax minimum value is not 10.")
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_OCC_2_1.py:125]:
```suggestion
            asserts.assert_greater_equal(hold_time_limits_dut.holdTimeMax, 10, "HoldTimeMax minimum value is not 10.")
            asserts.assert_greater_equal(hold_time_limits_dut.holdTimeMin, hold_time_limits_dut.holdTimeMax, "HoldTimeMax is not >= HoldTimeMin.")
```

File: src/python_testing/TC_OCC_2_1.py [hunk 2]
```diff
@@ -121,25 +121,31 @@ async def test_TC_OCC_2_1(self):
...
+            asserts.assert_less_equal(hold_time_limits_dut.holdTimeMin, 0xFFFE, "HoldTimeMin is not in valid range.")
+            asserts.assert_less_equal(hold_time_limits_dut.holdTimeMax, 0xFFFE, "HoldTimeMax is not in valid range.")
+            asserts.assert_greater_equal(hold_time_limits_dut.holdTimeMax, hold_time_limits_dut.holdTimeMin,
+                                         "HoldTimeMax is less than HoldTimeMin.")
             asserts.assert_less_equal(hold_time_limits_dut.holdTimeDefault,
-                                      hold_time_limits_dut.holdTimeMax, "HoldTimeMin is not in valid range")
+                                      hold_time_limits_dut.holdTimeMax, "HoldTimeDefault is greater than HoldTimeMax.")
             asserts.assert_greater_equal(hold_time_limits_dut.holdTimeDefault,
-                                         hold_time_limits_dut.holdTimeMin, "HoldTimeMin is not in valid range")
+                                         hold_time_limits_dut.holdTimeMin, "HoldTimeDefault is greater than HoldTimeMin.")
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_OCC_2_1.py:133]:
```suggestion
                                         hold_time_limits_dut.holdTimeMin, "HoldTimeDefault is less than HoldTimeMin.")
```

File: src/python_testing/TC_OCC_2_1.py [hunk 3]
```diff
@@ -121,25 +121,31 @@ async def test_TC_OCC_2_1(self):
...
         else:
             logging.info("HoldTimeLimits not supported. Test step skipped")
             self.mark_current_step_skipped()
 
         self.step(6)
         if attributes.HoldTime.attribute_id in attribute_list:
+            asserts.assert_in(attributes.HoldTimeLimits.attribute_id, attribute_list,
+                              "HoldTimeLimits attribute conformance failed.")
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_OCC_2_1.py:141]: This will never fail since the `if` above is the same condition.

**Review by tcarmelveilleux — APPROVED** [review]

**Review by tcarmelveilleux — APPROVED** [review]

**Review by leorozendaal — APPROVED** [review]

File: src/python_testing/TC_OCC_2_1.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tcarmelveilleux) [reply src/python_testing/TC_OCC_2_1.py:125]: Good catch

**Review by kiel-apple — APPROVED** [review]

**Pull request merged**